core/trino-parser/src/main/java/io/trino/sql/tree/DereferenceExpression.java

```diff
@@ -107,6 +127,11 @@ public static Expression from(QualifiedName name)
         return result;
     }
 
+    public static boolean isAllRowsReference(Expression expression)
```

Comment:
This doesn't seem like the right name. What does "all rows reference" mean? If it's referring to `xxx.*`, then that's a "qualified all fields reference". The qualifier could either be a label or a table name -- dependent on semantic context.